sdk/keyvault/Azure.Security.KeyVault.Keys/src/DeletedKey.cs

```diff
@@ -3,27 +3,66 @@
 // license information.
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Text.Json;
 
 namespace Azure.Security.KeyVault.Keys
 {
     public class DeletedKey : Key
     {
         public string RecoveryId { get; private set; }
 
-        public DateTime? DeletedDate { get; private set; }
+        public DateTimeOffset? DeletedDate { get; private set; }
 
-        public DateTime? ScheduledPurgeDate { get; private set; }
+        public DateTimeOffset? ScheduledPurgeDate { get; private set; }
 
-        public DeletedKey(string name, string recoveryId, DateTime? deletedDate, DateTime? scheduledPurge)
+        public DeletedKey(string name) : base(name) { }
+
+        public DeletedKey (string name, string recoveryId, DateTimeOffset? deletedDate, DateTimeOffset? scheduledPurge)
             : base(name)
         {
             RecoveryId = recoveryId;
             DeletedDate = deletedDate;
             ScheduledPurgeDate = scheduledPurge;
         }
+
+        internal override void WriteProperties(ref Utf8JsonWriter json)
+        {
+            base.WriteProperties(ref json);
+
+            if (RecoveryId != null)
+            {
+                json.WriteString("recoveryId", RecoveryId);
+            }
+
+            if (DeletedDate.HasValue)
+            {
+                json.WriteNumber("deletedDate", DeletedDate.Value.ToUnixTimeMilliseconds());
+            }
+
+            if (ScheduledPurgeDate.HasValue)
+            {
+                json.WriteNumber("scheduledPurgeDate", ScheduledPurgeDate.Value.ToUnixTimeMilliseconds());
```

Comment:
Who's  responsible for flushing the writer?

Reply:
The main Serialize under Model: https://github.com/Azure/azure-sdk-for-net/pull/6457/files#diff-11837fe870317cfc8557a2012f2e523cR33